torch/ao/quantization/quantizer/qnnpack_quantizer.py

```diff
@@ -0,0 +1,145 @@
+from torch.ao.quantization.backend._qnnpack_pt2e import get_qnnpack_pt2e_backend_config
+
+from .quantizer import Quantizer
+
+import typing
+import copy
+from dataclasses import asdict, dataclass
+from typing import Callable, Tuple, Union
+
+import torch
+
+SUPPORTED_DTYPES = [torch.uint8, torch.int8, torch.int32]
+SUPPORTED_QSCHEMES = [
+    torch.per_tensor_affine,
+    torch.per_channel_affine,
+    torch.per_channel_symmetric,
+    torch.per_channel_affine_float_qparams,
+]
+
+
+@dataclass
+class TensorSpec:
+    dtype: torch.dtype
+    quant_min: int
+    quant_max: int
+    qscheme: torch.per_tensor_affine
+    ch_axis: int
+    is_dynamic: bool = False
+
+    def __post_init__(self):
+        # check dtype is one of the supported types
+        if self.dtype not in SUPPORTED_DTYPES:
+            raise TypeError(f"Unsupported dtype {self.dtype}.")
+
+        # quant_min must be less than quant_max
+        if self.quant_min > self.quant_max:
+            raise ValueError(
+                f"quant_min {self.quant_min} must be <= quant_max {self.quant_max}."
+            )
+
+        # check qscheme is on of the supported ones
+        if self.qscheme not in SUPPORTED_QSCHEMES:
+            raise ValueError(f"Unsupported qscheme {self.qscheme}.")
+
+        # ch_axis must be less than the number of channels
+        # but no way to check here. Just check that it is not < 0.
+        if self.ch_axis < 0:
+            raise ValueError("Ch_axis is < 0.")
+
+
+OperatorSpec = typing.NamedTuple(
```

Comment:
Why OperatorSpec? I think QuantizationSpec is also good. Although given that this in quantization codebase, it does not have to be have Quantization in the name, but not sure what OperatorSpec means for user. For implementation it makes sense because you are talking about aten op.

Reply:
yeah this is just something as a starting point, please feel free to modify to something makes more sense to qnnpack users, I think the current naming mainly reflects the granularity of the spec, one is for tensor, the other is for operator

Reply:
I dont think we should distinguish between qnnpack users or other backends. I dont want to block this PR but all these have to be in base quantizer